### PR TITLE
Add paginate and use it in institutions

### PR DESCRIPTION
**Pull request by vintagepenguin**
This adds pagination for the .list method. 

@ztoolson

### CODE REVIEW

File: lib/atrium/paginate.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        # Add new objects to the list
+        response["#{klass_name}"].map do |params|
+          list << self.new(params)
+        end
+        @current_page += 1
+      end
+      list
+    end
+
+    def set_klass_name
```
Comment (ztoolson) [inline lib/atrium/paginate.rb]: why doesn't we just make this method klass_name? so then you don't have to call set_klass_name then use klass_name. you can just use it

Comment (vintagepenguin) [inline lib/atrium/paginate.rb]: 🙌

File: lib/atrium/institution.rb [hunk 2]
```diff
@@ -17,21 +16,12 @@ def self.credentials(institution_code)
...
     end
 
-    def self.list(options = nil)
-      endpoint = format_endpoint("institutions", options)
-      response = ::Atrium.client.make_request(:get, endpoint)
-
-      response["institutions"].map do |institution|
-        ::Atrium::Institution.new(institution)
-      end
+    def self.list(query_params: nil, limit: nil )
```
Comment (ztoolson) [inline lib/atrium/institution.rb]: nit picky but whitespace in the method. should be `def self.list(query_params: nil, limit: nil)`

**Comment by ztoolson** [comment]
other than my whitespace comment. let's 🚢

File: lib/atrium/institution.rb (hunk 2 above)
Comment (vintagepenguin) [inline lib/atrium/institution.rb]: @ztoolson you are the rubocop ⚡️

**Comment by ztoolson** [comment]
:shipit:

**Pull request merged**